tests/integration_tests/test_dataset.py

```
@@ -25,6 +25,47 @@ def test_construct_text_corpus():
     dataset.construct_text_corpus()
     dataset.close()
 
+def test_change_data_size():
```

This is a good test. One thing I'd suggest is to also test your failure cases. That is, I see you have a local function called `check_valid` in there. You should try to trigger that as well. IMO you should just raise a `ValueError`, and use an `assertRaises`. Check https://stackoverflow.com/a/61017213

---

I dont really know what this mean, perhaps clarify so i can get a better understanding. I did add two tests that check the inputs that are tokenized and not tokenized under the generator and not the generator.

---

Yup.

So basically here: https://github.com/opentensor/bittensor/pull/979/files#diff-3ba712d25bdb72a19a0b805fa92e1f6167b76383a937684d92fee4017b59846fR493
you define a function called `check_valid`, but none of your test cases will hit it.

That is, you're not passing in a `size <= 0` or a size that isn't an `int`. You should add test cases for these edge cases as well.

My other point was this block:

```
        if check_valid(batch_size):
            self.batch_size = batch_size
        
        if check_valid(block_size):
            self.block_size = block_size
```

I would argue that if *either* your batch size or block size is invalid to throw an error, rather than silently only setting one of them. e.g. if I passed in (10, -100), i should raise an error, rather than setting the batch_size to 10 and persisting the same block_size (especially cause the logger call in there indicates "success", when one of them, at least imo, failed).